src/analysis/special_functions/non_integrable.lean

```diff
@@ -0,0 +1,152 @@
+  (hd : ∀ᶠ x in l, differentiable_at ℝ f x) (hf : tendsto (λ x, ∥f x∥) l at_top)
+  (hg : is_O (deriv f) g l) :
+  ¬interval_integrable g volume a b :=
+begin
+  intro hgi,
+  have : ∃ (C : ℝ) (hC₀ : 0 ≤ C) (s ∈ l), ∀ (x ∈ s) (y ∈ s), [x, y] ⊆ [a, b] ∧
+    ∀ z ∈ [x, y], differentiable_at ℝ f z ∧ ∥deriv f z∥ ≤ C * ∥g z∥,
+  { rcases hg.exists_nonneg with ⟨C, C₀, hC⟩,
+    have h : ∀ᶠ x : ℝ × ℝ in l.prod l, ∀ y ∈ [x.1, x.2], (differentiable_at ℝ f y ∧
+      ∥deriv f y∥ ≤ C * ∥g y∥) ∧ y ∈ [a, b],
+      from (tendsto_fst.interval tendsto_snd).eventually ((hd.and hC.bound).and hl).lift'_powerset,
+    rcases mem_prod_self_iff.1 h with ⟨s, hsl, hs⟩,
+    simp only [prod_subset_iff, mem_set_of_eq] at hs,
+    exact ⟨C, C₀, s, hsl, λ x hx y hy, ⟨λ z hz, (hs x hx y hy z hz).2,
+      λ z hz, (hs x hx y hy z hz).1⟩⟩ },
+  clear hl hd hg,
+  choose C hC₀ s hsl hsub hfd hg,
+  replace hgi : interval_integrable (λ x, C * ∥g x∥) volume a b, by convert hgi.norm.smul C,
+  obtain ⟨c, hc, d, hd, hlt⟩ : ∃ (c ∈ s) (d ∈ s), ∥f c∥ + ∫ y in Ι a b, C * ∥g y∥ < ∥f d∥,
+  { rcases filter.nonempty_of_mem hsl with ⟨c, hc⟩,
+    have : ∀ᶠ x in l, ∥f c∥ + ∫ y in Ι a b, C * ∥g y∥ < ∥f x∥,
+      from hf.eventually (eventually_gt_at_top _),
+    exact ⟨c, hc, (this.and hsl).exists.imp (λ d hd, ⟨hd.2, hd.1⟩)⟩ },
+  specialize hsub c hc d hd, specialize hfd c hc d hd,
+  replace hg : ∀ x ∈ Ι c d, ∥deriv f x∥ ≤ C * ∥g x∥, from λ z hz, hg c hc d hd z ⟨hz.1.le, hz.2⟩,
+  have hg_ae : ∀ᵐ x ∂(volume.restrict (Ι c d)), ∥deriv f x∥ ≤ C * ∥g x∥,
+    from (ae_restrict_mem measurable_set_interval_oc).mono hg,
+  have hsub' : Ι c d ⊆ Ι a b,
+    from interval_oc_subset_interval_oc_of_interval_subset_interval hsub,
+  have hfi : interval_integrable (deriv f) volume c d,
+    from (hgi.mono_set hsub).mono_fun' (ae_measurable_deriv _ _) hg_ae,
+  refine hlt.not_le (sub_le_iff_le_add'.1 _),
+  calc ∥f d∥ - ∥f c∥ ≤ ∥f d - f c∥ : norm_sub_norm_le _ _
+  ... = ∥∫ x in c..d, deriv f x∥ : congr_arg _ (integral_deriv_eq_sub hfd hfi).symm
+  ... = ∥∫ x in Ι c d, deriv f x∥ : norm_integral_eq_norm_integral_Ioc _
+  ... ≤ ∫ x in Ι c d, ∥deriv f x∥ : norm_integral_le_integral_norm _
+  ... ≤ ∫ x in Ι c d, C * ∥g x∥ :
+    set_integral_mono_on hfi.norm.def (hgi.def.mono_set hsub') measurable_set_interval_oc hg
+  ... ≤ ∫ x in Ι a b, C * ∥g x∥ :
+    set_integral_mono_set hgi.def (ae_of_all _ $ λ x, mul_nonneg hC₀ (norm_nonneg _))
+      hsub'.eventually_le
+end
+
+/-- If `f` is differentiable in a punctured neighborhood of `c`, `∥f x∥ → ∞` as `x → c` (more
+formally, along the filter `𝓝[≠] c`), and `f' = O(g)` along `𝓝[≠] c`, where `f'` is the derivative
+of `f`, then `g` is not interval integrable on any nontrivial interval `a..b` such that
+`c ∈ [a, b]`. -/
+lemma not_interval_integrable_of_tendsto_norm_at_top_of_deriv_is_O_punctured {f : ℝ → E} {g : ℝ → F}
+  {a b c : ℝ} (h_deriv : ∀ᶠ x in 𝓝[≠] c, differentiable_at ℝ f x)
```

Comment:
This statement can not be applied if you only have assumptions to the left of `c`, or to the right of `c`, because the neighborhood `𝓝[≠] c` is two-sided. What about using instead the neighborhood within `[a, b] \ {c}` in the assumptions, which is strictly more general? Same thing below.

Reply:
If you only have estimate on one side (and you know which side), then you can apply the previous lemma.

Reply:
Of course you can always apply the previous lemma. But my point is that the current lemma, which is a specialization of the previous one, can be strictly improved, by putting the assumptions over `𝓝[[a, b] \ {c}] c` instead of `𝓝[≠] c`: as the former is smaller than the latter, the stronger assumptions imply the weaker ones, while the conclusion would be the same. If you prefer, you can even have the two specialized versions, deducing the `𝓝[≠] c` from the `𝓝[[a, b] \ {c}] c` version.

Reply:
Done.